app/DAO/BakeryDatabase.scala

```diff
@@ -68,41 +78,26 @@ class BakeryDatabase @Inject() (
     }
   }
 
-  def getProductById(id: String): Future[Option[String]] = {
-    Future {
-      db.withConnection { conn =>
-        try {
-          val statement = {
-            conn.prepareStatement("SELECT * FROM product WHERE id::text = ?")
-          }
-          statement.setString(1, id)
-
-          println(statement.toString)
-          val selectResult = statement.executeQuery()
-          val result = if (selectResult.next()) {
-            Some(
-              "id: " + selectResult.getString(
-                "id"
-              ) + ", name: " + selectResult.getString(
-                "name"
-              ) + ", quantity: " + selectResult.getInt(
-                "quantity"
-              ) + ", price: " + selectResult.getDouble(
-                "price"
-              ) + ", created at: " + selectResult.getTimestamp(
-                "created_at"
-              ) + ", updated at: " + selectResult.getTimestamp(
-                "updated_at"
-              )
-            )
-          } else {
-            None
-          }
-          result
-        } catch {
-          case e: Exception => Option(e.printStackTrace().toString)
-        }
-      }
-    }
+  def getProductById(id: String): Future[Option[Product]] = {
+    val query =
+      sql"""SELECT * FROM product where id::text = $id"""
+        .query[Product]
+        .option
+    query
+      .transact(bakeryTransactor.xa)
+      .unsafeToFuture()
```

Comment:
So the reason I gave the previous suggestion is because you were printing the query. If you don't want that anymore for debugging/ logging purposes, just chain the calls and skip the `val` declaration

Reply:
Since I'm not logging anymore, I cleaned up the query into one 'line'